### PR TITLE
Remove unsupported If-Match header value for Blob Immutability Policy operations

### PR DESCRIPTION
**Pull request by AbhishekKrishna123**
# Update to resource-manager / Microsoft.Storage
This change fixes an incorrect description in the spec which does not match the actual API implementation.
It has been incorrect since earlier, but has been fixed for the current latest stable API version.

### CODE REVIEW

File: specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json
Comment (blueww) [inline specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: Per the descritpion, is there a breaking change of the behavior on server (omit if-Match, before will apply the operation, now will fail?)
Besides that, if the format is still \"*\", please don't miss it in the updated description.

Comment (AbhishekKrishna123) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: Hi @blueww, thanks for the review. This PR is only a description change for correctness, and there is no change in server behavior.

The current description in our documentation is wrong. We have never supported "*" as a valid value on the server, so we are removing it.
Also, the if-match has always been required for Update operations. The "required" field is "false" for GET and "true" for PUT and other updates.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: @AbhishekKrishna123 

Thanks for the reply!

For `\"*\"`, I think it means there should be double quota in the begin/end of etag. 
If this is not the proper way to show that, it's better describe it in another way, but not remove the info from the description.

I am good with other part of the description change.

Comment (AbhishekKrishna123) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: @blueww We never supported "*" for the eTag value in our service, it is not related to the quotes. The documentation has been wrong since many years when the feature was introduced. Adding @HimanshuChhabra from my team for a review form our side.

Comment (HimanshuChhabra) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: The description change looks good to me, SRP does not support * or "*" values for e-tag.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: @AbhishekKrishna123 

I believe the * here is a wild char, `\"*\"` is to show the etag need have begin/end double quota.
It's OK for me to remove `\"*\"` , however we should let customer know etag need begin/end double quota in the description, to avoid customer failure.

Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: > I just reconfirmed that the etag value that we are returning from the service includes double quotes and the double quotes is required for the request to succeed. For example: ""8dca4aced6027dd"",

Yes, so the description should still indicate the double quotes need be in included in the etag (as the description before this PR indicate that with `\"*\"`).

Comment (AbhishekKrishna123) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json:547]: Sure, I will update the PR

Comment (AbhishekKrishna123) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json]: @blueww Please check if this change looks good. Added "The ETag value must include the leading and trailing double quotes as returned by the service."

**Review by blueww — APPROVED** [review]
LGTM

File: specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json
Comment (blueww) [reply specification/storage/resource-manager/Microsoft.Storage/stable/2023-05-01/blob.json]: Thanks for the update!
Looks good to me.
I have approved the PR.

**Comment by AbhishekKrishna123** [comment]
@yifanz7 You are marked as a Code Owner, could you please take a look and let me know if anything else is required from my end to merge this change? Thanks

**Comment by AbhishekKrishna123** [comment]
@yifanz7 You are marked as a Code Owner, could you please take a look and let me know if anything else is required to merge this change? Thanks

**Review by yifanz7 — APPROVED** [review]

**Comment by raosuhas** [comment]
> /pr RequestMerge

Can you please merge to main before requesting merge ?

**Comment by AbhishekKrishna123** [comment]
> > /pr RequestMerge
> 
> Can you please merge to main before requesting merge ?

@raosuhas I do not have write/merge permissions to merge it myself. I'm not sure what I'm missing here, can you please help? 
I followed the steps here: [aka.ms/azsdk/specreview/merge](https://aka.ms/azsdk/specreview/merge)

**Comment by AbhishekKrishna123** [comment]
> > > /pr RequestMerge
> > 
> > 
> > Can you please merge to main before requesting merge ?
> 
> @raosuhas I do not have write/merge permissions to merge it myself. I'm not sure what I'm missing here, can you please help? I followed the steps here: [aka.ms/azsdk/specreview/merge](https://aka.ms/azsdk/specreview/merge)

Hey @blueww, sorry to tag you here again, but is there anything else I need to do before requesting a merge here? The checks are passing (and few are not running), and reviewers have approved.

**Comment by blueww** [comment]
> > > > /pr RequestMerge
> > > 
> > > 
> > > Can you please merge to main before requesting merge ?
> > 
> > 
> > @raosuhas I do not have write/merge permissions to merge it myself. I'm not sure what I'm missing here, can you please help? I followed the steps here: [aka.ms/azsdk/specreview/merge](https://aka.ms/azsdk/specreview/merge)
> 
> Hey @blueww, sorry to tag you here again, but is there anything else I need to do before requesting a merge here? The checks are passing (and few are not running), and reviewers have approved.

I see the ARMSignedOff and you have already add RequestMerge tag.
Normally you can just wait for the PR merge.

**Comment by blueww** [comment]
@raosuhas 

Could you please raise if anything blocks the PR merge?

**Review by raosuhas — APPROVED** [review]

**Pull request merged**
